src/validation-middleware.ts

```diff
@@ -90,13 +94,13 @@ export const validationMiddleware = <H, P, Q, B, F, R>(
 
     await next();
 
-    const output = await validate(ctx.body, validation.response, 'response');
+    const output = await validate<R>(ctx.body, validation.response, 'response');
 
     if (!output) {
       return;
     }
 
-    if (output.error instanceof ZodError) {
+    if (assertZodValidationError<R>(output)) {
```

Comment:
is this change needed here, it doesn't seem to relate to adding the errors to ctx? is the type no longer a ZodError?

Reply:
It's needed to allow the generic type `<R>`, seems like that cannot be done with `instanceof ZodError<R>`, furthermore, the `assertZodValidationError` does the same, so I felt like it would be much better to unify this check.

Reply:
makes sense, its been awhile since i've worked on this :) 

thanks Tomas!